frame/composable-traits/src/dex.rs

```diff
@@ -101,6 +105,88 @@ pub trait Amm {
 	) -> Result<Self::Balance, DispatchError>;
 }
 
+/// Pool Fees
+#[derive(
+	Encode, Decode, MaxEncodedLen, TypeInfo, Clone, Default, PartialEq, Eq, Copy, RuntimeDebug,
+)]
+pub struct Fee<AssetId, Balance> {
+	// total fee
+	pub fee: Balance,
+	/// Amount of the fee pool charges for the exchange, this goes to liquidity providers.
+	pub lp_fee: Balance,
+	/// Amount of the fee that goes out to the owner of the pool
+	pub owner_fee: Balance,
+	/// Amount of the protocol fees(for PBLO holders) out of owner_fees.
+	pub protocol_fee: Balance,
+	/// assetId of the fees
+	pub asset_id: AssetId,
+}
+
+impl<AssetId: AssetIdLike, Balance: BalanceLike> Fee<AssetId, Balance> {
+	pub fn zero(asset_id: AssetId) -> Self {
+		Fee {
+			fee: Balance::zero(),
+			lp_fee: Balance::zero(),
+			owner_fee: Balance::zero(),
+			protocol_fee: Balance::zero(),
+			asset_id,
+		}
+	}
+}
+
+/// Pool Fee Config
+#[derive(
+	Encode, Decode, MaxEncodedLen, TypeInfo, Clone, Default, PartialEq, Eq, Copy, RuntimeDebug,
+)]
+pub struct FeeConfig {
+	/// Amount of the fee pool charges for the exchange, this goes to liquidity provider.
+	pub fee_rate: Permill,
+	/// Amount of the fee that goes out to the owner of the pool
+	pub owner_fee_rate: Permill,
+	/// Amount of the protocol fees(for PBLO holders) out of owner_fees.
+	pub protocol_fee_rate: Permill,
+}
+
+impl FeeConfig {
+	pub fn zero() -> Self {
+		FeeConfig {
+			fee_rate: Permill::zero(),
+			owner_fee_rate: Permill::zero(),
+			protocol_fee_rate: Permill::zero(),
+		}
+	}
+
+	pub fn calculate_fees<AssetId: AssetIdLike, Balance: BalanceLike>(
+		&self,
+		asset_id: AssetId,
+		amount: Balance,
+	) -> Fee<AssetId, Balance> {
+		let fee: Balance = self.fee_rate.mul_floor(amount);
+		let owner_fee: Balance = self.owner_fee_rate.mul_floor(fee);
+		let protocol_fee: Balance = self.protocol_fee_rate.mul_floor(owner_fee);
+		Fee {
+			fee,
+			// safe as the values are calculated as per million
+			lp_fee: fee - owner_fee,
+			owner_fee: owner_fee - protocol_fee,
+			protocol_fee,
+			asset_id,
+		}
+	}
+}
+
+impl Mul<Permill> for FeeConfig {
+	type Output = Self;
+
+	fn mul(self, rhs: Permill) -> Self::Output {
```

Comment:
Needs a test

Reply:
IIUC this is required due to add_liquidity() from StableSwap ?
I think if we just change fee_rate then that is enough right? I mean for purpose of StableSwap that is enough.
In that case better not have this mul() implemenation rater just fee_rate locally in add_liquidity()

Reply:
> IIUC this is required due to add_liquidity() from StableSwap ?

yeah.

> I think if we just change fee_rate then that is enough right?

Yeah, there is a mistake in the code here. But I wonder why there are no tests that fail for this stuff. I will add. About the `Mul` implementation, it's just to make the code cleaner. The idea is to allow increasing or decreasing the fee rate this way.